redpanda/templates/_helpers.tpl

```diff
@@ -404,4 +375,12 @@ IP is required for the advertised address.
     -}}
   {{- end -}}
 {{ $command | join " " }}
+{{- end -}}
+
+{{- define "storage-min-free-bytes" -}}
+{{- if dig "enabled" false .Values.storage.persistentVolume -}}
+  {{- mulf (include "SI-to-bytes" .Values.storage.persistentVolume.size) 0.1 | int64 -}}
```

Comment:
Maybe put an upper bound on it?  Reserving 500MB on a 5GB drive is probably sensible, but reserving 1.6TB of space on a 16TB drive is wasteful

Reply:
For context: the level of slack needed is more about bandwidth than capacity: we need to notice the issue and stop writes, in time to avoid actually filling the drive.  So on a system doing 1GB/s of writes and checking its disk state every 10 seconds (this should really be more frequent but that's what it is today), we need ~10GB of safety room.

Probably not realistic to calculate this dynamically, but that's the motivation for suggesting to used a <10% number on large drives.

Reply:
The default being 5 gig made me guess conservatively. The user can override that setting using the `config.cluster.storage_min_free_bytes` value. Would you suggest a more logarithmic calculation for this or do you think setting a conservative default and allowing users to override that be sufficient?